src/arrayMethodReduce.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator;
+    let startIndex;
+
+    if (arguments.length > 1) {
+      if (initialValue === undefined && this.length === 0) {
+        throw new TypeError('Reduce of empty array with no initial value');
```

Comment:
Actually, there shouldn't be any errors. The original reduce method works with an empty array and no initial value